DOCUMENTATION.md

```diff
@@ -0,0 +1,170 @@
+# Documentation
```

Comment:
This doc is pretty informative and I really appreciate it. 

No pressure in doing the following, but I think we should be able to auto-generate something like this based on just the docstring. I just tried pydoc-markdown, and it seems pretty feasible. The slight caveat there is that we would need to tweak a config file to exclude some of the private modules. This also gives us a chance to clean up the docstring. Maintaining the doc would also be easier in the ideal scenario.

Reply:
Thanks.

I've used Sphinx, readthedocs, before successfully, and I agree autogeneration is the way to go, long term. I've pre-emptively claimed torchsde on readthedocs -- not a priority right now but something to aim for on the next release, I think?

Reply:
SGTM. You could also add me to the readthedocs project, the username is the same as my github handle.

Reply:
Will do!